packages/router-plugin/src/core/router-generator-plugin.ts

```diff
@@ -15,7 +15,7 @@ const setLock = (bool: boolean) => {
 const PLUGIN_NAME = 'unplugin:router-generator'
 
 export const unpluginRouterGeneratorFactory: UnpluginFactory<
-  Partial<Config> | undefined
+  ConfigOptions | undefined
 > = (options = {}) => {
   let ROOT: string = process.cwd()
   let userConfig = options as Config
```

Comment:
This type casting here is risky, `Config` is the `z.output` type, which will contain defined properties due to the default option, but what's being passed it hasn't been parsed by Zod yet.

For example, `userConfig.routesDirectory` may not have been defined yet, so the usage on L24-26 is unsafe.

There is some complexity here, because we can't immediately parse the provided schema, as there may be later config retrieved from a `tsr.config.ts` file ([here](https://github.com/TanStack/router/blob/main/packages/router-generator/src/config.ts#L60)), and the default values would override anything provided in the config file